src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,120 @@
-import React from 'react';
+import React, { FormEvent, useCallback, useState } from 'react';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+import { ErrorType } from '../types/ErrorType';
+import { customFetch } from '../utils/fetchClient';
+import { validateEmailAddress } from '../helper/helper';
+import { ErrorForm } from '../types/ErrorTypesForm';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  post: Post | null,
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>,
+  setErrorType: React.Dispatch<React.SetStateAction<ErrorType | null>>,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  post,
+  setComments,
+  setErrorType,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [message, setMessage] = useState('');
+  const [hasName, setHasName] = useState(true);
+  const [hasEmail, setHasEmail] = useState(true);
+  const [isEmailValid, setIsEmailValid] = useState(true);
+  const [hasComment, setHasComment] = useState(true);
+  const [isCommentAdd, setIsCommentAdd] = useState(false);
+
+  const addNewComment = useCallback(async () => {
+    setIsCommentAdd(true);
```

Comment:
The useCallback is not correctly used. It should have all its dependencies specified in the array at the end. You are missing 'setComments' and 'setErrorType'.